src/CineFiles.jl

```diff
@@ -2,19 +2,219 @@ module CineFiles
 
 using ColorTypes: N0f8, N4f12, Gray
 using LRUCache
+using StaticArrays
 
-import Base: eltype, length, size, getindex, firstindex, lastindex, iterate
+import Base: eltype, length, size, getindex, firstindex, lastindex, iterate, read
 export CineFile
 
 # using FileIO
 # add_format(format"CINE", [0x43, 0x49], ".cine")
 
-struct Time64
+abstract type BinaryData end
+
+read(f::IOStream, S::Type{T}) where {T<:BinaryData} = S(read.(Ref(f), S.types)...)
+read(f::AbstractString, S::Type{T}) where {T<:BinaryData} = S(read.(f, S.types)...)
```

Comment:
What's the purpose of the `AbstractString` method? Why broadcast over the string?

Reply:
That can go. Its just in there from when I was debugging something and forgot to remove it.